health/notifications/alarm-notify.sh.in

```diff
@@ -939,7 +939,7 @@ send_email() {
     fi
 
     [ -n "${sender_email}" ] && opts+=(-f "${sender_email}")
-    [ -n "${sender_name}" ] && sendmail --help 2>&1 | grep -q "\-F " && opts+=(-F "${sender_name}")
+    [ -n "${sender_name}" ] && [[ ! `readlink "$(command -v sendmail 2>/dev/null)"` == */busybox ]] && opts+=(-F "${sender_name}")
```

Comment:
This is indeed an improvement, but is far from rigorous. Depending on how the system is set up, a system using busybox might alternatively use hardlinks, or even a one-line shellscript for each file.

A more robust approach would be to use `strings "$(command -v sendmail 2>/dev/null)" | grep -qi busybox`. That will correctly catch the case of a symlink (because `strings` will dereference the symlink), as well as the other two possibilities.

Reply:
Nice! But a note: On my system (Gentoo), sendmail is installed as:

`-r-xr-s--x 1 root smmsp 781856 Jun 25 13:36 /usr/sbin/sendmail`

Trying strings from a non-root user (which netdata might be running as) gives me:

```
evas@6am ~ # strings "$(command -v sendmail 2>/dev/null)" | grep -qi busybox
strings: /usr/sbin/sendmail: Permission denied
```

Reply:
Ah, yeah, I forgot about that case...

Unfortunately that makes it even more complicated, because we can’t detect the shell script case at all, and need to compare inode numbers with the busybox executable to catch the hardlink case.

Luckily, the shellscript case is relatively rare, but the hardlink case is not (it’s rather popular for tight embedded systems where space is at a true premium because it requires even less space than symlinks).

Reply:
There is always my ~~lame~~ approach - using a random option that results in printing usage + grep. I just had to use `-anything` instead of `--anything` 😄

Reply:
Perhaps we can combine both... :-) ?

Reply:
Okay, maybe we could just run `sendmail -F` and check the output.

On real sendmails, it should return `sendmail: option requires an argument -- F`.

but on like in our docker sendmail it will return: `sendmail: unrecognized option: F`

So, we can just grep for `requires an argument` .... ?

Reply:
Looks like an ok workaround 👍 😄

Reply:
Yep, that sounds like a sensible approach to me.